networking_aci/plugins/ml2/drivers/mech_aci/common.py

```diff
@@ -137,6 +137,24 @@ def get_hosts_on_network(self, context, network_id, level=None, with_segment=Fal
                 hosts.add((host, entry.segment_id))
             else:
                 hosts.add(host)
+
+        # find all segments in this network that belong to a transit (unbound segments)
+        if transit_hostgroups:
+            transit_physnets = [hg['physical_network'] for hg in transit_hostgroups]
+            query = context.session.query(segment_models.NetworkSegment.id,
+                                          segment_models.NetworkSegment.physical_network)
+            query = query.filter_by(network_type='vlan', network_id=network_id)
+            query = query.filter(segment_models.NetworkSegment.physical_network.in_(transit_physnets))
+            for entry in query.all():
+                # find host(group) of physnet
+                for hg in transit_hostgroups:
+                    if entry.physical_network == hg['physical_network']:
+                        break
+                if with_segment:
+                    hosts.add((host, entry.segment_id))
+                else:
+                    hosts.add(host)
```

Comment:
I think this shouldn't be `host` as `host` is just the last thing from the previous for-loop in line 135. I don't know what it should be, but probably the hosts on that segment we queried out ... not sure how those transits are named.

Reply:
`hg['hosts'][0]`, tnx